src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -15,148 +18,198 @@ namespace Microsoft.AspNetCore.Http.RequestDelegateGenerator.StaticRouteHandlerM
 
 internal class EndpointParameter
 {
-    public EndpointParameter(Endpoint endpoint, IParameterSymbol parameter, WellKnownTypes wellKnownTypes)
+    public EndpointParameter(Endpoint endpoint, IParameterSymbol parameter, WellKnownTypes wellKnownTypes): this(endpoint, parameter.Type, parameter.Name, wellKnownTypes)
     {
-        Type = parameter.Type;
-        SymbolName = parameter.Name;
-        LookupName = parameter.Name; // Default lookup name is same as parameter name (which is a valid C# identifier).
         Ordinal = parameter.Ordinal;
-        Source = EndpointParameterSource.Unknown;
         IsOptional = parameter.IsOptional();
         DefaultValue = parameter.GetDefaultValueString();
-        IsArray = TryGetArrayElementType(parameter, out var elementType);
+        ProcessEndpointParameterSource(endpoint, parameter, parameter.GetAttributes(), wellKnownTypes);
+    }
+
+    private EndpointParameter(Endpoint endpoint, IPropertySymbol property, IParameterSymbol? parameter, WellKnownTypes wellKnownTypes) : this(endpoint, property.Type, property.Name, wellKnownTypes)
+    {
+        Ordinal = parameter?.Ordinal ?? 0;
+        IsOptional = property.IsOptional() || parameter?.IsOptional() == true;
+        DefaultValue = parameter?.GetDefaultValueString() ?? "null";
+        // Coalesce attributes on the property and attributes on the matching parameter
+        var attributeBuilder = ImmutableArray.CreateBuilder<AttributeData>();
+        attributeBuilder.AddRange(property.GetAttributes());
+        if (parameter is not null)
+        {
+            attributeBuilder.AddRange(parameter.GetAttributes());
+        }
+        ProcessEndpointParameterSource(endpoint, property, attributeBuilder.ToImmutable(), wellKnownTypes);
+    }
+
+    private EndpointParameter(Endpoint endpoint, ITypeSymbol typeSymbol, string parameterName, WellKnownTypes wellKnownTypes)
```

Comment:
Is this only called from the ctor above?

Reply:
Correct, this contains the "core" implementation of the EndpointParameter that parameters sourced directly from the handler like so:

```
app.MapGet("/{name", (string name) => ...);
```

And those that are resolved as properties from an `AsParameters` attributed type resolve from:

```
app.MapGet("/{name", ([AsParameters] MyArgs args) => ...);

public class MyArgs
{
  string Name { get; set; }
}
```

Reply:
From the outside, the logical boundary isn't all that clear and it looks like a candidate for inlining.  If it's clear, in context, why it should be a separate method, then no objection.